interface/patient_file/summary/demographics_full.php

```diff
@@ -797,6 +810,54 @@ function end_group() {
     $("select").change(function() {
         checkSkipConditions();
     });
+
+//This code deals with demographics before save action -
+	<?php if ($hook):?>
+
+		var f = $("form");
+
+		// Use hook to open the controller and get the new patient validation .
+		// when no params are sent this window will be closed from the zend controller.
+		var url ='<?php echo  $GLOBALS['web_root']."/interface/modules/zend_modules/public/patientvalidation";?>';
+		$("#submit_btn").attr("type","button");
+		$("#submit_btn").attr("name","btnSubmit");
+		$("#submit_btn").attr("id","btnSubmit");
+		$("#btnSubmit").click(function( event ) {
+			if(!submitme(<?php echo $GLOBALS['new_validate'] ? 1 : 0;?>,event,'DEM',constraints)){
+				event.preventDefault();
+				return;
+			}
+			somethingChanged = false;
+			<?php
+			// D in edit_options indicates the field is used in duplication checking.
+			// This constructs a list of the names of those fields.
+			$mflist = "";
+			$mfres = sqlStatement("SELECT field_id FROM layout_options " .
+				"WHERE form_id = 'DEM' AND uor > 0 AND field_id != '' AND " .
+				"edit_options LIKE '%D%' " .
+				"ORDER BY group_name, seq");
+			while ($mfrow = sqlFetchArray($mfres)) {
+				$field_id  = $mfrow['field_id'];
+				if (strpos($field_id, 'em_') === 0) continue;
+				if (!empty($mflist)) $mflist .= ",";
+					$mflist .= "'" . text($field_id) . "'";
+			} ?>
+
+			var flds = new Array(<?php echo $mflist; ?>);
+			var separator = '?';
+			for (var i = 0; i < flds.length; ++i) {
+				var fval = $('#form_' + flds[i]).val();
+				if (fval && fval != '') {
+					url += separator;
+					separator = '&';
+					url += 'mf_' + flds[i] + '=' + encodeURIComponent(fval);
+				}
+			}
+			url += '&closeBeforeOpening=1';
+			dlgopen(url, '_blank', 700, 500);
+		});
+
+	<?php endif;?>
```

Comment:
To avoid a potential heart attack on your end, I'll bring this in after I just do the 1st step above. The only limiting factor at this point is finding the time to rebase this into one commit (should be able to do this tonight barring any on call craziness on my day job). I will then do above step 2 after it's in the codebase (this will be a nice simple exercise for me to try coding in Zend).